Move product editor utils to product editor package

joshuatf opened the pull request:
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

Moves all utils used in the new product editing experience to the `@woocommerce/product-editor` package.  Other components and hooks will come in follow-up PRs.

Closes #36719  .

<!-- The next section is mandatory. If your PR doesn't require testing, please indicate that you are purposefully omitting instructions. -->

- [ ] This PR is a very minor change/addition and does not require testing instructions (if checked you can ignore/remove the next section).

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Otherwise, please include detailed instructions on how these changes can be tested (including pre-conditions, configuration, steps to take and expected results). It may help to write your instructions using pseudocode -- as if you're telling a computer how to execute the test. -->

1. Navigate to the new product editing experience.
2. Generally smoke test the product editor and make sure it continues to work as expected

<!-- End testing instructions -->

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?

<!-- Mark completed items with an [x] -->

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

joshuatf committed:
Move product editor utils to product editor package

joshuatf committed:
Add changelog entries

joelclimbsthings commented on packages/js/product-editor/src/index.ts:
```diff
@@ -2,3 +2,25 @@ export {
+/**
+ * Utils
+ */
+export { formatCurrencyDisplayValue } from './utils/format-currency-display-value';
```
Likely subjective, but an alternative here would be to add an `index.ts` file within the `/utils` directory to export all utilities, and then potentially just `export * from './utils'`. Makes it a little smoother to import these utils within the package itself, since you can import all from `../utils`.

joelclimbsthings reviewed (approved):
Thanks for getting this done @joshuatf ! Gave the editor a pretty good smoke test, and all works well. I added one minor comment, but it's totally optional, so feel free to :shipit: .

joelclimbsthings reviewed (changes requested):
@joshuatf I also noticed that the original utils (such as `getCheckboxTracks`) haven't been removed, and they're actually still in use by various fills and such. Were you planning to divide this into a separate PR for some reason?

joshuatf committed:
Move remaining utils

joshuatf committed:
Move util import/exports to separate index file

joshuatf commented:
Thanks for taking a look at this, @joelclimbsthings!  It's been a few weeks since I opened this PR, so I'm not sure why I left the section utils separate, but I went ahead and moved them over.

Also moved the util import/exports to a separate file.

joelclimbsthings reviewed (approved):
Thanks Josh, changes look good and still tests well. 🚢
